src/geo_key_directory.rs

```diff
@@ -0,0 +1,647 @@
+                        &double_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::ProjScaleAtCenter => {
+                    directory.proj_scale_at_center = Self::get_double(
+                        &double_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::ProjAzimuthAngle => {
+                    directory.proj_azimuth_angle = Self::get_double(
+                        &double_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::ProjStraightVertPoleLong => {
+                    directory.proj_straight_vert_pole_long = Self::get_double(
+                        &double_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::Vertical => {
+                    directory.vertical =
+                        Self::get_short(key_tag, location_tag, *count, *value_or_offset)?.into()
+                }
+                GeoKeyDirectoryTag::VerticalCitation => {
+                    directory.vertical_citation = Self::get_string(
+                        &ascii_params_data,
+                        key_tag,
+                        location_tag,
+                        *count,
+                        *value_or_offset,
+                    )?
+                    .into()
+                }
+                GeoKeyDirectoryTag::VerticalDatum => {
+                    directory.vertical_datum =
+                        Self::get_short(key_tag, location_tag, *count, *value_or_offset)?.into()
+                }
+                GeoKeyDirectoryTag::VerticalUnits => {
+                    directory.vertical_units =
+                        Self::get_short(key_tag, location_tag, *count, *value_or_offset)?.into()
+                }
+            }
+        }
+
+        Ok(directory)
+    }
+
+    fn get_short(
+        key_tag: GeoKeyDirectoryTag,
+        location_tag: Option<Tag>,
+        count: u16,
+        offset: u16,
+    ) -> TiffResult<u16> {
+        if location_tag.is_some() {
+            return Err(TiffError::FormatError(TiffFormatError::Format(format!(
+                "Key `{key_tag:?}` did not have the expected value type."
+            ))));
+        }
```

Comment:
This check and error message is a little confusing to me at first glance. Would it be clearer if the match was something more explicit like `if location_tag is Tag::GeoDoubleParamsTag or Tag::GeoAsciiParamsTag`?

Reply:
Agree.

Reply:
Nice, thanks for the explanation. If we could add an inline-comment to say that this if-check is to ensure that `TIFFTagLocation == 0`, that would be great.